### PR TITLE
Generate credentials from accessTokens.json when user login azure cli with service principal

### PR DESCRIPTION
**Pull request by Flanker32**
Generate credentials from accessTokens.json when user login azure cli with service principal, for issue #125

### CODE REVIEW

**Comment by jdneo** [comment]
Please update the documents as well.

**Comment by Flanker32** [comment]
> Please update the documents as well.

Updated

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureAuthHelper.java [hunk 1]
```diff
@@ -240,7 +241,13 @@ protected Authenticated getAuthObjFromAzureCli() {
                 auth = azureConfigurable.authenticate(new MSICredentials());
             } else {
                 getLog().info(AUTH_WITH_AZURE_CLI);
-                auth = azureConfigurable.authenticate(AzureCliCredentials.create());
+                final AzureCliCredentials azureCliCredentials = AzureCliCredentials.create();
+                if (azureCliCredentials.clientId() != null) {
+                    return azureConfigurable.authenticate(azureCliCredentials);
+                } else {
+                    return azureConfigurable
+                            .authenticate(getCredentialFromAzureCliWithServicePrincipal());
```
Comment (jdneo) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureAuthHelper.java]: What will happen if `getCredentialFromAzureCliWithServicePrincipal()` returns `null`

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureAuthHelper.java [hunk 2]
```diff
@@ -250,6 +257,33 @@ protected Authenticated getAuthObjFromAzureCli() {
...
+     *
+     * @return Authenticated object if Azure CLI 2.0 is logged with Service Principal.
+     */
+    protected ApplicationTokenCredentials getCredentialFromAzureCliWithServicePrincipal() throws IOException {
+        final JsonObject subscription = getDefaultSubscriptionObject();
+        final JsonArray tokens = getAzureCliTokenList();
+        final String servicePrincipalName = subscription == null ? null : subscription.get("user")
+                .getAsJsonObject().get("name").getAsString();
+        if (servicePrincipalName == null || tokens == null) {
+            return null;
```
Comment (jdneo) [inline azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureAuthHelper.java]: Print some logs when returning null

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureAuthHelper.java (hunk 1 above)
Comment (Flanker32) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureAuthHelper.java]: Added null check

File: azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureAuthHelper.java (hunk 2 above)
Comment (Flanker32) [reply azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/AzureAuthHelper.java]: Done.

**Pull request merged**